### PR TITLE
(Github Actions) Bump versions of actions/checkout in ci workflow and scan for GitHub Actions with dependabot

### PR DESCRIPTION
**Pull request by ArjunSharda**
## Description

This pull request will bump the versions of actions/checkout in the ci workflow and for dependabot adds a package ecosystem for scanning for github actions.

### CODE REVIEW

**Comment by lildude** [comment]
Rejecting this as it has already been tried before and failed so had to be reverted. See https://github.com/github/linguist/pull/5915 

There is also no good reason to change this.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
     name: Classifier cross-validation
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
```
Comment (Zerocoolooog) [inline .github/workflows/ci.yml:52]: https://github.com/github-linguist/linguist/blob/master/.github%2Fworkflows%2Fci.yml

**Review by Zerocoolooog — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Alhadis) [reply .github/workflows/ci.yml:52]: > https://github.com/github-linguist/linguist/blob/master/.github%2Fworkflows%2Fci.yml

Please stop doing that.

Comment (Zerocoolooog) [reply .github/workflows/ci.yml:52]: > https://github.com/github-linguist/linguist/# blob/- [ ] @[]()WDR16BPBridgingtheDisabilityDividethroughDigitalTechnologyRAJA.pdf

- - @ master/.github%2Fworkflows%2Fci.yml

**Pull request closed without merging**